modules/WFCORE-3684_JDK Module Support.adoc

```diff
@@ -0,0 +1,94 @@
+= Title
+:author:            David Lloyd
+:email:             david.lloyd@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+In Java 9, the platform classes are divided up into modules; the core set being:
+
+[cols="3*m"]
+|======
+|java.base|java.compiler|java.datatransfer
+|java.desktop|java.instrument|java.jnlp
+|java.logging|java.management|java.management.rmi
+|java.naming|java.prefs|java.rmi
+|java.scripting|java.security.jgss|java.security.sasl
+|java.smartcardio|java.sql|java.sql.rowset
+|java.xml|java.xml.crypto|
+|======
+
+In addition, the ``java.se`` module re-exports many of these.
+
+As a first step towards alignment, we need to
+introduce the ability for modules to create module dependencies on
+these module names.  However, unless we want to require Java 9 or later from
+now on, the names must also work on Java 8.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3705[WFCORE-3705]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/MODULES-254[MODULES-254]
+* https://issues.jboss.org/browse/WFBUILD-37[WFBUILD-37]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* (pending)
+
+=== Affected Projects or Components
+
+* JBoss Modules
+* Selected WildFly Core modules
+* WildFly Build Tools
+
+=== Other Interested Projects
+
+* WildFly Swarm
+
+== Requirements
+
+=== Hard Requirements
+
+* Support all standard OpenJDK 9.x platform module names as built-in modules
+* Support ``org.jboss.modules`` as a built-in module
+* Allow the boot module loader to load platform modules like ``java.logging`` or ``jdk.sctp``
+** When running on Java 8, emulate these modules as built-in system classpath aggregations
+** Exclude modules which are not present on Java 11+ such as ``java.transaction`` and ``java.corba``
```

Comment:
Eventually adding a list of these modules could be useful (for documentation purposes, if anything).

Reply:
I've added a list of these modules to the bottom of the doc.